filters/apiusagemonitoring/pathinfo.go

```diff
@@ -16,8 +17,9 @@ type pathInfo struct {
 	CommonPrefix   string
 	ClientPrefix   string
 
-	metricPrefixesPerMethod [methodIndexLength]*endpointMetricNames
-	metricPrefixedPerClient map[string]*clientMetricNames
+	metricPrefixesPerMethod [methodIndexLength]*endpointMetricNames // endpoint metric names cache
+	metricPrefixedPerClient map[string]*clientMetricNames           // client metric names cache
```

Comment:
Let us use `sync.Map` as replacement here?

The code to maintain will be shorter and easier to understand using a standard pattern, and there are smart guys that have thought about the performance implications of locks and formed it into this code.

Reply:
I would rather try to put the endpointMetricNames and the clientMetricNames into a single combined structure and protect it with a single RW lock scope, if the logic allows. As an alternative approach, I would even consider putting the scope of the Response() method of the filter into a single shared RW lock scope. The reason why the latter may be better, is that the endpointMetricNames is always accessed, and this way the code can be safer for future changes. For performance optimizations, we should create benchmarks first.

Reply:
sync.Map is not simpler to maintain. 
+1 for @aryszka suggestion